docs/madr/1-reassign-evolve.md

```diff
@@ -0,0 +1,166 @@
+if the APIs perform as expected - this is very inefficient as dry run calls to
+CU's are expensive in networking client side, as well as for the CU.
+
+## Decision Drivers
+
+- Improve the performance of ANT loading client side.
+- Ensure the upgrade process is more robust by using immutable versioning, such
+  as module IDs.
+- Reduce computational load on CU infra.
+- Mitigate risks associated with reassigning ANT state and ARNS records.
+
+## Considered Options
+
+1. **Current Approach (Baseline)**
+   - Multiple dry runs per ANT.
+   - ETH compatibility checked via handler analysis.
+   - Upgrade involves loading Lua code.
+2. **Leverage Re-Assign Name and \_boot capabilites to upgrade ANTs**
+   - Use the module ID to evaluate process version and supported capabilites
+   - Upgrade via spawning a new ANT and reassigning ARNS records.
+
+## Decision Outcome
+
+### Chosen Option: **Leverage Re-Assign Name and \_boot capabilites to upgrade ANTs**
+
+With the ARIO process now supporting
+[`Reassign-Name`](https://github.com/ar-io/ar-io-ant-process/pull/26) and recent
+additions to add [`_boot`](https://github.com/ar-io/ar-io-ant-process/pull/57)
+to fork the process state to a whole new ANT, we can assign that new ANT as the
+registered ANT to the specified ArNS name in the registry, and using the module
+ID of the process to identify process version and capabilities, and solely the
+module ID.
+
+This optimized approach reduces computational overhead and improves caching for
+better performance. It also shifts to a more robust upgrade mechanism using
+immutable module IDs rather than version detection heuristics. We can do this
+now that we compile and publish our own WASM binaries, rather than using an AOS
+binary and loading our code into it.
+
+```mermaid
+sequenceDiagram
+    autonumber
+
+    participant ARIOGateway
+    participant Owner
+    participant OldAnt
+    participant NewAnt
+    participant AntRegistry
+    participant ARIOProcess
+
+
+    Owner ->> ARIOGateway: Get Module ID for process from GQL
+    ARIOGateway -->> Owner: Module ID Result
+
+
+    alt No upgrade needed Module ID from process meta is up to date
+
+        else
+
+        Owner->>OldAnt: Get state
+        OldAnt-->>Owner: Return state
+
+        Owner->>NewAnt: Spawn new ant with old state
+        NewAnt->>AntRegistry: Register ANT notice
+        Owner->>NewAnt: Get State to verify responsiveness and Module ID to verify correct deployment
+
+        Owner->>OldAnt: Sign and accept Reassign message
+        OldAnt->>ARIOProcess: Forward Reassign Message
+        ARIOProcess-->>OldAnt: Reassign-Notice
+
+        Owner->>ARIOProcess: Check for Reassignment success and notify user
+
+
+    end
+```
+
+### Positive Consequences
+
+- **Performance Improvement:** Reducing dry runs significantly speeds up ANT
+  loading.
+- **Immutable Module Versioning:** Eliminates issues with heuristic version
```

Comment:
we discussed this and i do not see it,

by publishing a module and validating it's capabilities before reassinging a name, we can guarantee the new ANT works as expected AND has the latest features/fixes as described by the module. We could not do this before with evals

Reply:
I actually had it further down but added it here as well [59241af](https://github.com/ar-io/ar-io-ant-process/pull/74/commits/59241affbdf49e4f55853e7c811acf02db2343ab)